### DISCUSSION TITLE
Choosing a colour scheme based on ssh/mux

### DISCUSSION THREAD

**Discussion by joelgibson** [wezterm/wezterm#4138] [Q&A] [answered]
Hi, I've been trying out Wezterm (and it's awesome, I've been wanting something with proper multiplexing support for a while). A setup I find helpful is for my terminal colour scheme to change depending on whether I am logged into a remote box or not, so that I have an obvious visual indicator of where I am. I could do this by passing a different `--config` to `wezterm connect` I suppose, but is there a way to detect that I am in a mux when loading the lua config, and just use an if statement in the config?

Cheers, Joel.

**Comment by wez** [comment, marked as answer]
Take a look at:

* https://wezfurlong.org/wezterm/config/lua/window-events/update-status.html which fires periodically and when the tab/pane focus changes
* https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html which allows changing the config for the current window
* https://wezfurlong.org/wezterm/config/lua/pane/get_domain_name.html which you can use to see whether the pane is local or one of your remote mux panes

Something like:

```lua
local wezterm = require 'wezterm'

local DOMAIN_TO_SCHEME = {
  -- the keys correspond to your ssh and/or tls domain names
  ["my-domain"] = "Batman",
  ["my-other-domain"] = "cyberpunk",
}

wezterm.on('update-status', function(window, pane)
  local domain = pane:get_domain_name()

  -- show the domain name in the right status area to aid in debugging/understanding
  window:set_right_status(domain)

  local overrides = window:get_config_overrides() or {}
  -- resolve the scheme for the domain. If there is no mapping, then the overridden
  -- scheme is cleared and your default colors will be used
  overrides.color_scheme = DOMAIN_TO_SCHEME[domain]
  window:set_config_overrides(overrides)
end)
```

**Reply by joelgibson** [reply]
Thank you very much for the solution and quick reply - it works like a charm.

**Comment by aurels** [comment]
Inspired by this, I've done that to switch the theme if I'm in an SSH session : 

```lua
COLOR_SCHEME = 'Tokyo Night'
SSH_COLOR_SCHEME = 'Red Alert'

[...]

wezterm.on('update-status',
  function(window, pane)
    local fg_process_name = pane:get_foreground_process_name()

    local overrides = window:get_config_overrides() or {}

    if fg_process_name == '/usr/bin/ssh' then
      overrides.color_scheme =  SSH_COLOR_SCHEME
    else
      overrides.color_scheme = COLOR_SCHEME
    end

    window:set_config_overrides(overrides)
  end
)
```

Hoping it may help someone !
